app/routes/goal_routes.py

```diff
@@ -1 +1,83 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from .route_utilities import create_model, get_models_with_filters, validate_model
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    return create_model(Goal, request_body)
+
+@bp.get("")
+def get_all_goals():
+    return get_models_with_filters(Goal, request.args)
+
+@bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    result = {}
+    result["goal"] = goal.to_dict()
+    return result
```

Comment:
I'd opt for `response` since that is more descriptive of the data it is holding.
```suggestion
    response = {}
    response["goal"] = goal.to_dict()
    
    return response
```